desktop/cmp/treemap/TreeMapModel.js

```diff
@@ -0,0 +1,295 @@
+ * mapped to label (a node's display name), value (a node's size), and heat (a node's color).
+ *
+ * Can also (optionally) be bound to a GridModel. This will enable selection syncing and
+ * expand / collapse syncing for GridModels in `treeMode`.
+ *
+ * Color customization can be managed by setting colorAxis stops via the `highchartsConfig`.
+ * @see Dark and Light themes for examples.
+ */
+@HoistModel
+export class TreeMapModel {
+
+    //------------------------
+    // Immutable public properties
+    //------------------------
+    /** @member {Store} */
+    store;
+    /** @member {GridModel} */
+    gridModel;
+    /** @member {string} */
+    labelField;
+    /** @member {string} */
+    valueField;
+    /** @member {string} */
+    heatField;
+    /** @member {string} */
+    valueFieldLabel;
+    /** @member {string} */
+    heatFieldLabel;
+    /** @member {number} */
+    maxDepth;
+    /** @member {function} */
+    filter;
+    /** @member {function} */
+    onClick;
+    /** @member {function} */
+    onDoubleClick;
+    /** @member {string} */
+    algorithm;
+    /** @member {(boolean|TreeMapModel~tooltipFn)} */
+    tooltip;
+
+    //------------------------
+    // Observable API
+    //------------------------
+    /** @member {Object} */
+    @bindable.ref highchartsConfig = {};
+    /** @member {TreeMapRecord[]} */
+    @observable.ref data = [];
+
+    /**
+     * @param {Object} c - TreeMapModel configuration.
+     * @param {Store} [c.store] - A store containing records to be displayed.
+     * @param {GridModel} [c.gridModel] - Optional GridModel to bind to.
+     * @param {Object} [c.highchartsConfig] - Highcharts configuration object for the managed chart. May include
+     *      any Highcharts opts other than `series`, which should be set via dedicated config.
+     * @param {string} c.labelField - Record field to use to determine node label.
+     * @param {string} c.valueField - Record field to use to determine node size.
+     * @param {string} c.heatField - Record field to use to determine node color.
+     * @param {string} [c.valueFieldLabel] - Label for valueField to render in the default tooltip.
+     * @param {string} [c.heatFieldLabel] - Label for heatField to render in the default tooltip.
+     * @param {number} [c.maxDepth] - Maximum tree depth to render.
+     * @parma {function} [c.filter] - A filter function used when processing data. Receives (record), returns boolean.
+     * @param {function} [c.onClick] - Callback to call when a node is clicked. Receives (record, e).
+     *      If not provided, by default will select a record when using a GridModel.
+     * @param {function} [c.onDoubleClick] - Callback to call when a node is double clicked. Receives (record, e).
+     *      If not provided, by default will expand / collapse a record when using a GridModel.
+     * @param {string} [c.algorithm] - Layout algorithm to use. Either 'sliceAndDice', 'stripes', 'squarified' or 'strip'.
```

Comment:
I suggest we provide a link to https://www.highcharts.com/docs/chart-and-series-types/treemap in the top-level class comment, and could refer developers there in this doc string - otherwise these end up being pretty mysterious.  Would also note default of squarified.